[vro] (#297) Fix VcComputeResourceConfigSpec class and add its dependencies in vRO (#298)
vmware/build-tools-for-vmware-aria: docs/versions/latest/Release.md

[//]: # (Optional But higlhy recommended Specify *NONE* if missing)
[//]: # (#### Relevant Documentation:)

### Add missing properties to VcComputeResourceConfigSpec and related class definitions in vRO interfaces

#### Previous Behavior

> The VcComputeResourceConfigSpec class is missing a few attributes and types of those attributes

Reviewer: ```suggestion
The `VcComputeResourceConfigSpec` class is missing a few attributes and related class interfaces.
```

#### Current Behavior

> The attributes are missing and corresponding types are missing

Reviewer: Here is a suggestion on how we can describe the current / new behaviour:
```suggestion
The `enableConfigManager`, `maximumHardwareVersionKey` and `desiredSoftwareSpec` attributes are added to the `VcComputeResourceConfigSpec` class interface. Related class interfaces for `VcDesiredSoftwareSpec`, `VcDesiredSoftwareSpecBaseImageSpec`, `VcDesiredSoftwareSpecVendorAddOnSpec` and `VcDesiredSoftwareSpecComponentSpec` are added vCenter plugin interfaces.
```

Author: Got it.

> The attributes are missing and corresponding types are missing

Reviewer 2: I think the sentence should be:
`The missing attributes and corresponding types were added`
